wagtail/api/v2/serializers.py

```diff
@@ -40,7 +40,9 @@ def get_attribute(self, instance):
         return instance
 
     def to_representation(self, obj):
-        return type(obj)._meta.app_label + '.' + type(obj).__name__
+        name = type(obj)._meta.app_label + '.' + type(obj).__name__
+        self.context['view'].seen_types[name] = type(obj)
```

Comment:
Yep, I agree. There isn't a nice way around this at the moment. Serializers currently also need to be aware of other endpoints in order to support cross-linking between them.

I've started to address this in https://github.com/torchbox/wagtail/pull/2484. Instead of allowing serialisers to make nested serialisers, we would generate all serialisers at the view level (and pass nested serialisers into their parents). This would also mean that all the types used in the API would be available in the view so this bit of code wouldn't be necessary